### ISSUE TITLE
error initializing syslog stream ... io timeout ... pops up every 2 minutes

### ISSUE THREAD

**Issue by slaesh** [grafana/loki#5677]
**Describe the bug**
I keep getting this error like every 2 minutes:

```
promtail     | level=warn ts=2022-03-19T15:52:41.292609338Z caller=syslogtarget.go:206 msg="error initializing syslog stream" err="read tcp 172.20.0.2:1514->172.20.0.3:36993: i/o timeout"
```

**To Reproduce**

1. my docker-compose file
```yaml
version: "3"

services:
  syslog-ng:
    container_name: syslog-ng
    image: lscr.io/linuxserver/syslog-ng
    restart: unless-stopped
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Berlin
    volumes:
      - ./config:/config
      - ./logs:/var/log

    ports:
      - 5514:5514/udp
#      - 6601:6601/tcp
#      - 6514:6514/tcp

  promtail:
    container_name: promtail
    image: grafana/promtail
    restart: unless-stopped
    volumes:
      - /var/lib/docker/containers:/var/lib/docker/containers:ro
      - ./promtail-config.yml:/etc/promtail/promtail-config.yml
    command: -config.file=/etc/promtail/promtail-config.yml
    ports:
      - 1514:1514 # receive logs
      - 9080:9080 # web portal
```
2. the promtail config:
```yaml
server:
  http_listen_port: 9080
  grpc_listen_port: 0

positions:
  filename: /tmp/positions.yaml

client:
  url: https://<userid>:<token>@logs-prod-eu-west-0.grafana.net/loki/api/v1/push

scrape_configs:

- job_name: syslog
  syslog:
    listen_address: 0.0.0.0:1514
    idle_timeout: 60s
    label_structured_data: yes
    labels:
      job: "syslog"
  relabel_configs:
    - source_labels: ['__syslog_message_hostname']
      target_label: 'host'

```
3. syslog-ng config:
```yaml
@version: 3.13

source s_net {
    udp(port(5514));
};

destination d_file {
    file("/var/log/remote/${FULLHOST}");
};

destination d_loki {
    syslog("promtail" transport("tcp") port("1514"));
};

log {
    source(s_net);
    destination(d_file);
    destination(d_loki);
};

```
4. start it
5. push syslog messages to the system
6. they appear in the syslog-ng logs, they appear in the grafana-cloud! everything "seems" fine. :)
7. BUT, there are these errors/warnings keep spamming..

**Expected behavior**
no errors/warnings printed

**Environment:**
 - Infrastructure: laptop
 - Deployment tool: docker-compose

promtail version and start:

```
promtail     | level=info ts=2022-03-19T16:23:15.571705622Z caller=syslogtarget.go:108 msg="syslog listening on address" address=[::]:1514 tls=false
promtail     | level=info ts=2022-03-19T16:23:15.573711693Z caller=server.go:260 http=[::]:9080 grpc=[::]:39923 msg="server listening on addresses"
promtail     | level=info ts=2022-03-19T16:23:15.574371378Z caller=main.go:119 msg="Starting Promtail" version="(version=2.4.2, branch=HEAD, revision=525040a32)"
```

**Comment by chaudum** [comment]
Hey @slaesh 
Can you try with the newly released version 2.5.0 again and see if the error log still occurs?
There have been some changes recently that could have solved your issue.

**Comment by slaesh** [comment]
sweet! thanks, looks good!
